core/src/services/azdls/config.rs

```
@@ -58,6 +62,153 @@ pub struct AzdlsConfig {
+                    endpoint.set_protocol(value)?;
+                }
+                "EndpointSuffix" => {
+                    endpoint.set_suffix(value)?;
+                }
+                "SharedAccessSignature" => {
+                    config.sas_token = Some(value.to_string());
+                }
+                "BlobEndpoint" | "FileEndpoint" | "QueueEndpoint" | "TableEndpoint" => {
+                    // ADLSv2 endpoints need to be built up explicitly from the
+                    // `AccountName`, `EndpointSuffix` and `DefaultEndpointsProtocol`
+                    // fields instead. There is currently no equivalent for DFS.
+                    return Err(Error::new(
+                        ErrorKind::ConfigInvalid,
+                        format!("{} is not supported with Azdls", key),
+                    ));
+                }
+                "UseDevelopmentStorage" => {
+                    if value == "true" {
+                        use_development_storage = true;
+                    }
+                }
+                "DevelopmentStorageProxyUri" => {
+                    development_storage_proxy_uri = Some(value.to_string());
+                }
+                _ => {
+                    return Err(Error::new(
+                        ErrorKind::ConfigInvalid,
+                        format!("Unexpected key in connection string: {}", key),
+                    ));
+                }
+            }
+        }
+
+        if use_development_storage {
+            apply_azurite_parameters(&mut config, development_storage_proxy_uri)?;
+        } else {
+            endpoint.validate()?;
+            config.endpoint = Some(endpoint.to_string());
+        }
+
+        Ok(config)
+    }
+}
+
+/// Takes a semicolon-delimited Azure Storage connection string and returns
+/// key-value pairs split from it.
+fn parse_connection_string(conn_str: &str) -> Result<Vec<(&str, &str)>> {
+    conn_str
+        .split(';')
+        .map(|field| {
```

In https://github.com/apache/opendal/pull/5482/ we are going to introduce a `query_pairs` http util that does almost the same thing (but silently ignoring error instead of failing) 
https://github.com/apache/opendal/blob/7309b4bb46a2b8f3338fcf3646c9475fc3c7ee81/core/src/raw/http_util/uri.rs#L71

---

Thanks, this is great! I'll change it (and add a `separator: &str` parameter to accommodate the `;`) if your PR gets in first 👍